src/cli/commands/install.js

```diff
@@ -230,11 +230,43 @@ export class Install {
       }
 
       this.rootManifestRegistries.push(registry);
-      const json = await this.config.readJson(loc);
-      await normalizeManifest(json, this.config.cwd, this.config, true);
+      const projectManifestJson = await this.config.readJson(loc);
+      await normalizeManifest(projectManifestJson, this.config.cwd, this.config, true);
+
+      const rootCwd = this.config.cwd;
+      // if project has workspaces we aggreagate all dependences from workspaces into root
+      if (projectManifestJson.workspaces && this.config.workspacesExperimental) {
+        if (!projectManifestJson.private) {
+          throw new MessageError(this.reporter.lang('workspacesRequirePrivateProjects'));
+        }
+        for (const workspace of projectManifestJson.workspaces) {
+          for (const workspaceLoc of await fs.glob(path.join(rootCwd, workspace, filename))) {
```

Comment:
I've added a few utility methods to the `Config` class to support workspaces (for example https://github.com/yarnpkg/yarn/pull/3365/files#diff-9d9a6cd82f41984872a66a3ab0d440c4R555), it would make sense to update the code to use them once we've merged this PR, what do you think?